libsigsegv: disable stackoverflow tests on x86_64-darwin

xworld21 opened the pull request:
Disable tests that fail on Rosetta, preventing Hydra from building binaries for x86_64-darwin.

This *should* fix #327716 in a way that does not disable the tests completely: they should still run if building on a native x86 machine. However, I do not have an ARM64 machine to test that the `uname` trick has the desired effect, so somebody will need to confirm that this works as I hope.

## Things done

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin (native, not Rosetta)
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

veprbl commented on pkgs/by-name/li/libsigsegv/package.nix:
No, Rosetta builds are not necessarily cross builds.

Prince213 replied:
I see, so it's something we can only detect at build time.

xworld21 replied:
> I see, so it's something we can only detect at build time.

Yes, that would be my understanding of how Nixpkgs runs on Rosetta (i.e. it's the entire x86_64-darwin Nixpkgs as if it were running on an actual x86_64), and how Hydra can populate the x86_64 binary cache from M* machines, since as far as I know you cannot do that via cross compiling.

emilazy replied:
Yes, no point conditionalizing this when all our `x86_64-darwin` builders are Rosetta 2, especially given https://github.com/NixOS/nixpkgs/pull/415566.

xworld21 replied:
> As in: let’s not even bother with the `uname` stuff, just disable them.

My reasoning was to keep the checks for the vanishing number of people building libsigsegv on native x86... although I for one hope I won't have to build libsigsegv again.

donn reviewed (changes requested):
From my basic testing `uname -m` returns `x86_64` under Rosetta, including during nix builds: [Apple recommends](https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment#Determine-Whether-Your-App-Is-Running-as-a-Translated-Binary) `sysctl -n sysctl.proc_translated` to check if you're running under Rosetta:

`$(sysctl -n sysctl.proc_translated 2> /dev/null || echo '0')` will return 0 if native and 1 if running under Rosetta

```
[22:18] donn@Donns-Mac-mini:~% uname -m
x86_64
[22:18] donn@Donns-Mac-mini:~% sysctl -n sysctl.proc_translated
sysctl: unknown oid 'sysctl.proc_translated'
[22:18] donn@Donns-Mac-mini:~% exit
Connection to donns-mac-mini.local closed.
[22:18] donn@Donns-Mac-mini-M4:~% nix run nixpkgs#legacyPackages.x86_64-darwin.bash
[22:18] donn@Donns-Mac-mini-M4:~$ uname -m
x86_64
[22:18] donn@Donns-Mac-mini-M4:~$ sysctl -n sysctl.proc_translated
1
[22:18] donn@Donns-Mac-mini-M4:~$ exit
[22:19] donn@Donns-Mac-mini-M4:~% nix run nixpkgs#legacyPackages.aarch64-darwin.bash
[22:19] donn@Donns-Mac-mini-M4:~$ uname -m
arm64
[22:19] donn@Donns-Mac-mini-M4:~$ sysctl -n sysctl.proc_translated
0
```

xworld21 committed:
libsigsegv: disable stackoverflow tests on x86_64-darwin

Disable tests that fail on Rosetta, preventing Hydra from building
binaries for x86_64-darwin.

donn commented on pkgs/by-name/li/libsigsegv/package.nix:
```diff
@@ -16,6 +16,14 @@ stdenv.mkDerivation rec {
 
+  # stackoverflow tests fail on Rosetta, preventing Hydra from building x86_64-darwin binaries
+  preConfigure = lib.optionalString (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) ''
+    if [[ $(uname -m) != x86_64 ]] ; then
```
Thanks, pretty bad at shell if you couldn't tell. :) 

The patch does get applied on Rosetta now. Not entirely sure how the non-zero exit within the variable assignment is handled natively though. 

```
PASS: test-catch-segv1
PASS: test-catch-segv2
PASS: test-segv-dispatcher1
============================================================================
Testsuite summary for libsigsegv 2.15
============================================================================
# TOTAL: 3
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

---

Being honest though if @emilazy advises they be removed in all cases, she knows Nix on Darwin best and they should just be removed in all cases.
